lib/ruby_smb/smb2/file.rb

```diff
@@ -240,16 +250,24 @@ def set_header_fields(request)
       # @return [WindowsError::ErrorCode] the NTStatus code returned from the operation
       # @raise [RubySMB::Error::InvalidPacket] if the response is not a WriteResponse packet
       def write(data:'', offset: 0)
+        max_write = tree.client.server_max_write_size
+        if tree.client.dialect == '0x0202' || !tree.client.server_supports_multi_credit
+          max_write = 65536
+        end
         buffer            = data.dup
         bytes             = data.length
-        atomic_write_size = if bytes > tree.client.server_max_write_size
-                              tree.client.server_max_write_size
+        atomic_write_size = if bytes > max_write
+                              max_write
                             else
                              bytes
                             end
```

Comment:
I think using [Enumerable#min](https://ruby-doc.org/core-2.7.1/Enumerable.html#method-i-min) here would be a bit cleaner.

```suggestion
        atomic_write_size = [bytes, max_write].min
```

Reply:
Agreed